propagation/propagation.go

```diff
@@ -29,6 +29,15 @@ type TextMapCarrier interface {
 	// must never be done outside of a new major release.
 }
 
+// MultiTextMapCarrier is a TextMapCarrier that can return multiple values for a single key.
+type MultiTextMapCarrier interface {
+	TextMapCarrier
```

Comment:
Does this need to be embedded. It limits the composability.

Reply:
I was also thinking about two interfaces; one defining the method, second embedding two interfaces. Inspired that in `io` package there is e.g. https://pkg.go.dev/io#ReadCloser.